### ISSUE TITLE
Destroy where with empty array for ids

### ISSUE THREAD

**Issue by haches** [sequelize/sequelize#3132]
Using sequelize 2.0.2

When trying to delete files with the following code

```
var fileIdsToDelete = [];

return db.File
  .destroy({where: {id: fileIdsToDelete}});
```

it generates the invalid SQL
```
DELETE FROM `Files` WHERE `id` IN ()
```

and Sequelize crashes with "SequelizeDatabaseError: ER_PARSE_ERROR: You have an error in your SQL syntax;..."

This code used to work with older versions. Is the new behavior intentional?

**Comment by mickhansen** [comment]
No it appears to be a regression bug.

**Comment by ruddct** [comment]
I'm also seeing this in 2.0.2

**Comment by mickhansen** [comment]
Yes it's a general thing for where clauses'

**Comment by ruddct** [comment]
You're a rockstart @mickhansen

**Comment by haches** [comment]
Thanks @mickhansen. That was super fast.

**Comment by bulkan** [comment]
@mickhansen I don't believe 2.0.2 contains this fix ?

**Comment by ruddct** [comment]
@bulkan This was submitted after 2.0.2 was released, changelog says it'll be out in the next rev bump

**Comment by bulkan** [comment]
@ruddct awesome thanks. A work around for the moment is

```
Model.findAll({
    where: {
      id: {
        in: ids.length ? ids : [null]
      }
    }
})
```

**Comment by mickhansen** [comment]
Fixed in `sequelize@2.0.3`
